pkg/identityserver/application_registry.go

```diff
@@ -161,6 +168,9 @@ func (is *IdentityServer) listApplications(ctx context.Context, req *ttnpb.ListA
 			return nil, err
 		}
 	}
+	if req.IncludeDeleted {
+		ctx = store.WithSoftDeleted(ctx, false)
+	}
```

Comment:
With regards to the UX that I outlined above, what would be needed is actually an exclusive filter. Something like `IsDeleted` which then only gets deleted or not deleted records.